dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/test-applications/beanvalidation/src/com/ibm/ws/jaxrs20/cdi12/fat/beanvalidation/BookStoreWithValidationPerRequestLeak.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jaxrs20.cdi12.fat.beanvalidation;
+
+import java.util.WeakHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+@Path("/perrequestleak/")
+public class BookStoreWithValidationPerRequestLeak {
+
+    static WeakHashMap<BookStoreWithValidationPerRequestLeak, String> map = new WeakHashMap<BookStoreWithValidationPerRequestLeak, String>();
```

Comment:
```suggestion
    static WeakHashMap<BookStoreWithValidationPerRequestLeak, String> map = new WeakHashMap<>();
```
This is a nit, but might as well do it the simpler way that was added in Java 7.